### PR TITLE
Deploy MCM during Worker deletion

### PR DESCRIPTION
**Pull request by timuthy**
**What this PR does / why we need it**:
This PR (re)deploys MCM when when `Worker` resources are deleted. This is a precaution to ensure that existing worker nodes are deleted properly.

/cc @DockToFuture 

**Special notes for your reviewer**:
The timestamp in the controller logs have been adjusted to a human readable format.

**Release note**:

```improvement operator
The worker actuator does now re-apply the `machine-controller-manager` deployment during worker deletion to ensure a proper worker node deletion.
```

### CODE REVIEW

File: pkg/controller/worker/genericactuator/actuator_delete.go [hunk 1]
```diff
@@ -47,11 +44,13 @@ func (a *genericActuator) Delete(ctx context.Context, worker *extensionsv1alpha1
...
 	// Make sure machine-controller-manager is awake before deleting the machines.
-	deployment := &appsv1.Deployment{}
-	if err := a.client.Get(ctx, kutil.Key(worker.Namespace, a.mcmName), deployment); err != nil {
-		return err
+	var replicaFunc = func() (int32, error) {
+		return 1, nil
 	}
-	if err := util.ScaleDeployment(ctx, a.client, deployment, 1); err != nil {
+
+	// Deploy the machine-controller-manager into the cluster to make sure
```
Comment (rfranzke) [inline pkg/controller/worker/genericactuator/actuator_delete.go]: Missing doc string ending :-)

File: pkg/controller/worker/genericactuator/actuator_reconcile.go [hunk 2]
```diff
@@ -47,12 +47,33 @@ func (a *genericActuator) Reconcile(ctx context.Context, worker *extensionsv1alp
...
+	}
+
 	// Deploy the machine-controller-manager into the cluster.
 	a.logger.Info("Deploying the machine-controller-manager", "worker", fmt.Sprintf("%s/%s", worker.Namespace, worker.Name))
-	if err := a.deployMachineControllerManager(ctx, worker, cluster, workerDelegate); err != nil {
+	if err := a.deployMachineControllerManager(ctx, worker, cluster, workerDelegate, replicaFunc); err != nil {
 		return err
 	}
 
+	if !controller.IsHibernated(cluster.Shoot) {
```
Comment (rfranzke) [inline pkg/controller/worker/genericactuator/actuator_reconcile.go]: I think we don't need this `if` condition anymore. We have added it back then when the worker actuator was talking to the shoot's API server directly to apply the required RBAC roles for MCM. Now, it creates `ManagedResource`s and isn't talking to the shoot API server anymore.

File: pkg/controller/worker/genericactuator/machine_controller_manager.go [hunk 3]
```diff
@@ -48,29 +48,17 @@ func (a *genericActuator) deployMachineControllerManager(ctx context.Context, wo
...
 	}
+	mcmValues["replicas"] = replicaCount
 
 	if err := a.mcmSeedChart.Apply(ctx, a.chartApplier, workerObj.Namespace,
 		a.imageVector, a.gardenerClientset.Version(), cluster.Shoot.Spec.Kubernetes.Version, mcmValues); err != nil {
 		return errors.Wrapf(err, "could not apply MCM chart in seed for worker '%s'", util.ObjectName(workerObj))
 	}
 
-	if !controller.IsHibernated(cluster.Shoot) {
-		if err := a.applyMachineControllerManagerShootChart(ctx, workerDelegate, workerObj, cluster); err != nil {
```
Comment (rfranzke) [inline pkg/controller/worker/genericactuator/machine_controller_manager.go]: Why not keeping this here? I guess it is related to the deployment of MCM itself, so why did you move it to the actuator reconciliation function?

Comment (timuthy) [reply pkg/controller/worker/genericactuator/machine_controller_manager.go]: I moved it because of the `!controller.IsHibernated(cluster.Shoot)` condition. But since you said we didn't need it anymore, I'll move it back.
